Source/ARTLocalDevice.m

```diff
@@ -60,52 +57,35 @@ + (ARTLocalDevice *)load:(ARTRest *_Nonnull)rest {
     }
     device.push.recipient[@"transportType"] = ARTDevicePushTransportType;
 
-    NSString *deviceId = [rest.storage objectForKey:ARTDeviceIdKey];
+    NSString *deviceId = [storage objectForKey:ARTDeviceIdKey];
     if (!deviceId) {
-        deviceId = [[ULID new] ulidString];
-        [rest.storage setObject:deviceId forKey:ARTDeviceIdKey];
+        deviceId = [self generateId];
+        [storage setObject:deviceId forKey:ARTDeviceIdKey];
     }
     device.id = deviceId;
-    device.updateToken = [rest.storage objectForKey:ARTDeviceUpdateTokenKey];
 
-    [device setDeviceToken:[rest.storage objectForKey:ARTDeviceTokenKey]];
+    NSString *deviceSecret = [storage secretForDevice:deviceId];
+    if (!deviceSecret) {
+        deviceSecret = [self generateSecret];
+        [storage setSecret:deviceSecret forDevice:deviceId];
+    }
+    device.secret = deviceSecret;
+
+    device->_identityTokenDetails = [ARTDeviceIdentityTokenDetails unarchive:[storage objectForKey:ARTDeviceIdentityTokenKey]];
```

Comment:
Use a setter if needed. This feels a bit "hacky".

Reply:
It's the only way. `identityTokenDetails` prop is readonly.